src/rts/UnitAction.java

```diff
@@ -733,7 +737,7 @@ public static void getValidActionArray(Unit u, GameState gs, UnitTypeTable utt,
                 case TYPE_ATTACK_LOCATION: {
                     int relative_x = ua.x - u.getX();
                     int relative_y = ua.y - u.getY();
-                    mask[idxOffset+6+4+4+4+4+numUnits+(centerCoordinate+relative_y)*maxAttackRange+(centerCoordinate+relative_x)] = 1;
+                    mask[idxOffset+6+4+4+4+4+numUnitTypes+(centerCoordinate+relative_y)*maxAttackRange+(centerCoordinate+relative_x)] = 1;
```

Comment:
I've just appended a new commit where I replaced the `+6` cases by `NUMBER_OF_ACTION_TYPES`. Each of the `+4` cases is for the four cardinal directions, but a different "flavour" of directions every time (once for movement directions, once for harvest directions, once for resource-returning-to-base directions...). Instead of replacing all those by variables with long names, I added comment lines above them briefly explaining what's going on.

Reply:
Ah, thanks! That sounds good. I just wanted to make sure that the "6" was not for "number of unit types" or something like that, which is bound to change depending of the UnitTypeTable :)